yarn-plugin/sources/utils.ts

```diff
@@ -80,13 +159,37 @@ export const getSortedPackages = async (
   return packages;
 };
 
-export interface ManifestWithLicenseInfo {
-  name: string;
-  license?: ManifestLicenseValue;
-  licenses?: ManifestLicenseValue | ManifestLicenseValue[];
-  repository?: { url: string } | string;
-  homepage?: string;
-  author?: { name: string; url: string };
+export async function getPackageInfos(
+  packageManifest: ManifestWithLicenseInfo,
+): Promise<string> {
+  const { repository } = packageManifest;
+  const formattedRepository: string = getDownloadLocation(repository);
+  return formattedRepository;
 }
 
-type ManifestLicenseValue = string | { type: string };
+export function getDownloadLocation(
+  repository: { url: string } | string | undefined,
+): string {
+  if (
+    repository &&
+    typeof repository === "object" &&
+    isValidDownloadLocation(repository.url)
```

Comment:
if it is an `object` but the download location is not valid it probably should fail or return `undefined`?

Reply:
I have no strong opinion here. For now, I fixed it as suggested in https://github.com/spdx/tools-ts/pull/127